src/util/records.py

```diff
@@ -309,6 +309,43 @@ def remove_asset(cr, name):
         remove_view(cr, name, silent=True)
 # fmt:on
 
+def get_models_inheriting_actions_actions(cr):
+    cr.execute(
+        """
+        SELECT child_model.model
+            FROM ir_model AS child_model
+            JOIN ir_model_inherit
+            ON child_model.id = ir_model_inherit.model_id
+            JOIN ir_model AS base_model
+            ON ir_model_inherit.parent_id = base_model.id
+            WHERE base_model.model = 'ir.actions.actions'
+        """
+    )
+    return cr.fetchall()
+
+def remove_action(cr, xml_id=None, action_id=None):
+    assert bool(xml_id) ^ bool(action_id)
+    
+    if xml_id:
+        action_id = ref(cr, xml_id)
+
+    if action_id:
+        module, _, name = xml_id.partition(".")
+        cr.execute("SELECT model FROM ir_model_data WHERE module=%s AND name=%s", [module, name])
+        [model] = cr.fetchone()
+        if not model in get_models_inheriting_actions_actions(cr):
+            raise ValueError(
+                "%r should point to a model inheriting from 'ir.actions.actions', not a %r" % (xml_id, model)
+            )
+    
+    cr.execute("SELECT name, model FROM ir_model_fields WHERE relation = 'ir.actions.actions' AND on_delete ='cascade'")
```

Comment:
You should also search the one linking to the xmlid's model
```suggestion
    cr.execute("SELECT name, model FROM ir_model_fields WHERE relation IN %s AND on_delete ='cascade'", [tuple({model, "ir.actions.actions"})])
```